Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -491,6 +584,16 @@ public class NIOSSLHandler : ChannelInboundHandler, ChannelOutboundHandler, Remo
         }
     }
 
+    /// Checks if the `allowRemoteHalfClosure` channel option is set.
+    private func getAllowRemoteHalfClosureFromChannel(context: ChannelHandlerContext) -> Bool {
+        var halfClosureAllowed = false
+        if let syncOptions = context.channel.syncOptions {
+            let result = try? syncOptions.getOption(ChannelOptions.allowRemoteHalfClosure)
+            result.map { halfClosureAllowed = $0 }
```

Comment:
Using `map` for the side-effect is a bit weird here, it's probably better to use `if let`.